### ISSUE TITLE
Versioning

### ISSUE THREAD

**Issue by ealeksandrov** [MailOnline/ImageViewer#78]
Why 2-part version numbers are used? Why not classic [semantic versioning](http://semver.org)?

If there is no apparent reason to keep it - I suggest using full `MAJOR.MINOR.PATCH` in future releases:

`MAJOR` - breaking API change
`MINOR` - new features added
`PATCH` - bugfixes without breaking any public API

**Comment by ealeksandrov** [comment]
Looking at current version 4.0 - I'm not sure, if I should add it as `'~> 4.0'` to Podfile. Will 4.1 have any breaking change? Unknown. But I still want bugfixes - so I have to check changelog manually on each release.

**Comment by zfoltin** [comment]
We do use semantic versioning (kinda) with the caveat that zero patch numbers are left out for clarity. We try to release something when it's stable enough and won't require a patch. Obviously there can be bugs and issues but by the time they get discovered (and fixed) we might already have other changes that are waiting for release which will warrant a new minor version increase anyway.
See https://github.com/mojombo/semver/issues/237 for the discussion on this issue.

Hope this makes sense.

At the same time, I'm happy to append the zeros if that makes more people happy than unhappy :)

P.S. We will try to be more vigilant at documenting changes more clearly in a `CHANGELOG` - it's always good to look through that before upgrading and not relying solely on version numbers.

**Comment by ealeksandrov** [comment]
> We try to release something when it's stable enough and won't require a patch.
Awesome idea but nobody achieved it yet :)

I hear you. But from my experience sometimes after major release (lets say 2.1.0) bug reports are starting to come in and 2.1.1/2.1.2/... are needed to quickly fix bugs introduced.
So as a developer I am adding `~> 2.1.0` and I can be 99% sure that any updates will not break anything but only improve stability.

As an example of violation - https://github.com/rs/SDWebImage/issues/653. This hit me when it was updated while being a dependency of another lib and broke everything. I can't simply update API or change version manually, since it is inside code of another lib. I had to wait for fix or fork and fix or fix locally re-add libs manually.

**Comment by zfoltin** [comment]
Absolutely agree in terms of changing the major version number on breaking changes. It's just to me, changes in the patch version number VS changes in the minor version number doesn't make any difference from the consumer's perspective. When either changes I will assume no breaking changes, that's all. Probably when the minor version number changes then more people will go and look at the `CHANGELOG` to see what's new...

Anyways, until the semver guys decide (if ever) to make the patch number optional, I'm happy to add it to new releases going forward. 👍

**Comment by ealeksandrov** [comment]
Funny that this was closed. Looking back half of year later - seems that this library won't use any sort of versioning in any near future.

I have to cut this from connected Pods, include directly as a source, mark "legacy" and forget about using it in any future projects.
It can't be relied upon - no changelogs, no patch versions, no any sort of versions - nothing.

I tried to help with this project - but my 3.1 patch never was deployed, it's still ugly note in Readme. Changelog where I added all changes carefully extracted from raw commits - never was amended or used as release notes.
